src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -570,19 +575,19 @@ calculateFloatingPaymentClaimsFromSwapStream floatingRateCalculation s
           floatingRateCalculation.floatingRateIndex floatingRateSpread rateFixingCalendar fixLeg
           paymentPeriodData []
         where
-          (notional, paymentPeriodData) = calculateNotionalAndFixings p s
+          (Some notional, paymentPeriodData) = calculateNotionalAndFixings p s
             calculationScheduleStaticData fxRateId useAdjustedDatesForDcf periodicSchedule
 
       -- calculate simple interest rate claim (no calc period compounding) for a payment period
       -- using a non-compounded reference rate
       createSimpleClaim p =
         mconcat $ map createSimpleRate paymentPeriodData
         where
-          (notional, paymentPeriodData) = calculateNotionalAndFixings p s
+          (_, paymentPeriodData) = calculateNotionalAndFixings p s
             calculationScheduleStaticData fxRateId useAdjustedDatesForDcf periodicSchedule
           createSimpleRate pp =
-            when (TimeGte pp.rateFixingDate) $ scale notional $ scale (rate * pp.dcf) $
-              when (TimeGte p.adjustedEndDate) $ one currency
+            when (TimeGte p.adjustedEndDate) $ scale pp.notional $ scale (rate * pp.dcf) $
```

Comment:
can be made a single `scale` node

Reply:
I tried that, but it causes rounding differences in many tests, e.g. 1592500.00014 instead of 1592500.0

Seems this means decreased accuracy? Also, it makes the nodes of the claims tree more difficult to interpret / debug.

Reply:
That is interesting, what is the desired result? I am keen to understand where the numerical difference is coming from

Reply:
I think 1592500.0 is the desired result: 100,000,000 * 0.063 * 91/360

Reply:
I have pushed https://github.com/digital-asset/daml-finance/pull/814/commits/6e8a79b36e22374ae77ccfdeb833c8cfdf287133 below if you want to have a look. I am referring to the first floating coupon date (line 2913) of the `runFpml4` test, but there are many others that are failing as well.

I think it is best to stick with two scale nodes for the time being.

Reply:
Following your analysis, I decided to stick to one scale node, enforcing the calculation order by:

`scale (pp.notional * periodRate)`